packages/web-app-app-store/src/views/AppList.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+  <div class="app-list oc-mb-m">
+    <h2 class="oc-mt-rm" v-text="$gettext('App Store')" />
+    <div class="oc-flex oc-flex-middle">
+      <oc-text-input
+        id="apps-filter"
+        v-model.trim="filterTerm"
+        :label="$gettext('Search')"
+        :clear-button-enabled="true"
+        autocomplete="off"
+      />
+    </div>
+    <oc-list class="app-tiles">
+      <app-tile
+        v-for="app in filteredApps"
+        :key="`app-${app.repository.name}-${app.id}`"
+        :app="app"
+        class="oc-my-m"
+        @search="setFilterTerm"
+      />
+    </oc-list>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, nextTick, onMounted, ref, unref, watch } from 'vue'
+import Mark from 'mark.js'
+import Fuse from 'fuse.js'
+import { useAppsStore } from '../piniaStores'
+import AppTile from '../components/AppTile.vue'
+import { storeToRefs } from 'pinia'
+import { App } from '../types'
+import { defaultFuseOptions } from '@ownclouders/web-pkg'
+
+export default defineComponent({
+  name: 'AppList',
+  components: { AppTile },
+  setup() {
+    const appsStore = useAppsStore()
+    const { apps } = storeToRefs(appsStore)
+
+    const filterTerm = ref('')
+    const setFilterTerm = (term: string) => {
+      filterTerm.value = term.trim()
+    }
+    const filter = (apps: App[], filterTerm: string) => {
+      if (!(filterTerm || '').trim()) {
+        return apps
+      }
+      const searchEngine = new Fuse(apps, {
+        ...defaultFuseOptions,
+        keys: ['name', 'subtitle', 'tags']
+      })
+      return searchEngine.search(filterTerm).map((r) => r.item)
+    }
+    const filteredApps = computed(() => {
+      // TODO: debounce the filtering by 100-300ms
+      return filter(unref(apps), unref(filterTerm)).sort((a, b) =>
+        a.name.toLowerCase().localeCompare(b.name.toLowerCase())
+      )
+    })
+
+    const markInstance = ref(null)
+    onMounted(async () => {
+      await nextTick()
+      markInstance.value = new Mark('.mark-element')
+    })
+    watch(filterTerm, () => {
+      unref(markInstance)?.unmark()
+      if (unref(filterTerm)) {
+        unref(markInstance)?.mark(unref(filterTerm), {
+          element: 'span',
+          className: 'mark-highlight'
+        })
+      }
```

Comment:
we have a little problem here, mark-highlight has a  font-weight of 600, which means that the app title will be smaller when 'marked'
![image](https://github.com/user-attachments/assets/8c2e1230-cf0a-4492-b410-b1965b3a5cc3)